src/python/gem5/resources/client_api/atlasclient.py

```diff
@@ -155,24 +156,39 @@ def _atlas_http_json_req(
 
     def get_resources(
         self,
-        resource_id: Optional[str] = None,
-        resource_version: Optional[str] = None,
-        gem5_version: Optional[str] = None,
-    ) -> List[Dict[str, Any]]:
+        client_queries: List[ClientQuery],
+    ) -> Dict[str, Any]:
         url = f"{self.url}/action/find"
         data = {
             "dataSource": self.dataSource,
             "collection": self.collection,
             "database": self.database,
         }
-        filter = {}
-        if resource_id:
-            filter["id"] = resource_id
-            if resource_version is not None:
-                filter["resource_version"] = resource_version
 
-        if filter:
-            data["filter"] = filter
+        search_conditions = []
+        for resource in client_queries:
+            condition = {
+                "id": resource.get_resource_id(),
+            }
+
+            if not resource.get_gem5_version().startswith("DEVELOP"):
```

Comment:
It'd be neater here to add a function in `ClientQuery` like `is_develop_version(self) -> bool`. First of all we do this comparison a few times, so best to have the logic in one place. Secondly, whether you're querying for develop or an official version, is a property of the query so the logic should be put there.

Reply:
Also, maybe I'm wrong but doesn't this break if `get_gem5_version()` returns `None`?

Reply:
`gem5_version` should never be `None` as by default it is the current version. I will change the default in client query class to point to the current gem5 version.